### DISCUSSION TITLE
Kitty scrolls down when output takes up whole screen

### DISCUSSION THREAD

**Discussion by vibhav1011** [kovidgoyal/kitty#3777] [Q&A] [answered]
[Only in fullscreen] When the output of a command is big and takes up the entire screen, writing a new command in kitty drives the entire output up and I have to scroll back and forth to see something form the output while typing a new command. 
Is there a way to modify this behaviour?

Expected result :
![image](https://user-images.githubusercontent.com/26878621/123853029-9a00b400-d93a-11eb-97d2-b77e0370843d.png)
To be able to type with the screen as is^.

What happens :
![image](https://user-images.githubusercontent.com/26878621/123853253-df24e600-d93a-11eb-84a9-b7f3acbd88f6.png)

**Comment by kovidgoyal** [comment]
That is something your shel would have to implement, not the terminal.

**Reply by vibhav1011** [reply]
But this doesn't happen with the xfce terminal using the same shell.

**Comment by kovidgoyal** [comment]
Then I dont understand what it is you are asking, can you explain some
more.

On Tue, Jun 29, 2021 at 10:28:33PM -0700, vibhav1011 wrote:
> But this doesn't happen with the xfce terminal using the same shell.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/3777#discussioncomment-941430

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by vibhav1011** [reply]
In the second image - the command line is scrolled up - I didn't do this, it happened automatically as soon as I started typing. 
What I would like to happen is to be able to type with the command line at the bottom of the screen after the output of xrandr. 

![image](https://user-images.githubusercontent.com/26878621/123926026-1d58ee80-d9a9-11eb-9719-c209ebe521a2.png)

**Reply by kovidgoyal** [reply]
I cannot replicate that, steps I tried:

1) Run xrandr without any arguments
2) Prompt appears at bottom of screen
3) typing at prompt it remains at bottom of screen

If you mean you are using xrandr to resize the screenand therefore the kitty window, use https://sw.kovidgoyal.net/kitty/conf.html#opt-kitty.scrollback_fill_enlarged_window  option to have kitty fill enlarged windows from the scrollback.

**Reply by vibhav1011** [reply]
xrandr was just used as an example because the output is large enough to warrant full screening. The issue is not xrandr specific.

So the steps I take to produce this behaviour are: 
1. Launch kitty (kitty launches in default 600x400 window)
2. run xrandr (or any other command that fills the entire kitty window)
3. maximise kitty (Super + up for me)
4. And the type prompt is already in the middle of the screen.
After this happens, even if I manually scroll up and then start typing the type prompt jumps to the middle of the screen.

**Comment by kovidgoyal** [comment, marked as answer]
Again, use the setting I linked too

**Reply by vibhav1011** [reply]
Thanks! This worked.
